scipy/sparse/_coo.py

```diff
@@ -556,16 +577,68 @@ def _add_dense(self, other):
         dtype = upcast_char(self.dtype.char, other.dtype.char)
         result = np.array(other, dtype=dtype, copy=True)
         fortran = int(result.flags.f_contiguous)
-        M, N = self._shape_as_2d
-        coo_todense(M, N, self.nnz, self.row, self.col, self.data,
-                    result.ravel('A'), fortran)
+        if self.ndim == 1:
+            coo_todense_nd(np.array([1]), self.nnz, self.ndim,
+                           self.coords[0], self.data, result.ravel('A'), fortran)
+        elif self.ndim == 2:
+            M, N = self._shape_as_2d
+            coo_todense(M, N, self.nnz, self.row, self.col, self.data,
+                        result.ravel('A'), fortran)
+        else:
+            shape = np.array(self.shape)
+            if fortran:
+                strides = np.append(1, np.cumprod(shape[:-1]))
+            else:
+                strides = np.append(np.cumprod(shape[1:][::-1])[::-1], 1)
```

Comment:
I guess this is the same as above....   
```suggestion
        else:
            if fortran:
                strides = np.append(1, np.cumprod(self.shape[:-1]))
            else:
                strides = np.append(np.cumprod(self.shape[1:][::-1])[::-1], 1)
```

Also, do you want me to add a commit to fix the linting (which is not part of what you've touched, but is close enough that ruff is checking and making suggestions...). Once I push it you would need to pull your branch to any local repo before making more changes.

Reply:
Yes, that sounds good, thanks!